src/mlpack/methods/ann/layer/softmax_impl.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file softmax_impl.hpp
+ * @author Mrityunjay Tripathi
+ * @author Sreenik Seal
+ *
+ * Implementation of the Softmax class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTMAX_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTMAX_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "softmax.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+SoftMax<InputDataType, OutputDataType>::SoftMax()
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void SoftMax<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  output.set_size(arma::size(input));
```

Comment:
Since you assign the output in line 37, no need to set the size here.

Reply:
Done but I wonder how armadillo does this? Is it not that when we initialize the matrix with predefined shape (static allocation), it accesses the elements faster?